packages/libraries/router/src/usage.rs

```diff
@@ -186,14 +194,25 @@ impl Plugin for UsagePlugin {
             return Err("Hive token is required".into());
         }
 
-        let endpoint = init
+        let mut endpoint = init
             .config
             .registry_usage_endpoint
             .clone()
             .unwrap_or_else(|| {
                 env::var("HIVE_ENDPOINT").unwrap_or(DEFAULT_HIVE_USAGE_ENDPOINT.to_string())
             });
 
+        let target_id = init
+            .config
+            .target
+            .clone()
+            .or_else(|| env::var("HIVE_TARGET_ID").ok());
+
+        // In case target ID is specified in configuration, append it to the endpoint
+        if let Some(target_id) = &target_id {
+            endpoint.push_str(&format!("/{}", target_id));
+        }
```

Comment:
![](https://media0.giphy.com/media/v1.Y2lkPTc5MGI3NjExZG1oaHhlazB1Y3BxM2RqMGUwMTQ4cjc3NWFyaDRmNzRidmZ3OTkxMiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/dZcPjnRLVUsEGR9MCo/giphy.gif)

Reply:
yeah sorry no real tests at the moment. will try to add a few next week if it's not a blocker